---
title: "ITS Soil Metagenomics"
author: "Nehemiah Ongeso"
date: "10/18/2022"
output: html_document
---
```{r}
#Load Packages
suppressPackageStartupMessages({
require(csv)
require(dplyr)
require(tidyr)
library(dada2)
library(Cairo)
library(ggtree)
library(VennDiagram)
library(UpSetR)
library("phyloseq"); packageVersion("phyloseq") # Handling and analysis of high-throughput microbiome census data.
library("vegan");packageVersion("vegan") # Community Ecology Package.
library("ggplot2");packageVersion("ggplot2") # Create Elegant Data Visualisations Using the Grammar of Graphics.
library("dendextend");packageVersion("dendextend")
library("tidyr");packageVersion("tidyr")
library("viridis");packageVersion("viridis")
library("reshape");packageVersion("reshape")
#install.packages("remotes")
#remotes::install_github("vmikk/metagMisc")
library(metagMisc)
library(janitor)
library(coin) # Conditional Inference Procedures in a Permutation Test Framework.
library(reshape2) # Flexibly Reshape Data: A Reboot of the Reshape Package. 
library(ggnewscale) # Multiple Fill and Colour Scales in 'ggplot2'.
library(MicrobiotaProcess) # an R package for analysis, visualization and biomarker discovery of Microbiome.
library(patchwork)
})
```

```{r}
rm(list=ls())
#loading the ASV taxonomic table
getwd()
tax<-read.table("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/ASV_tax_species.tsv", header = T, row.names = 1,sep = "\t")
head(tax)
names(tax)
tax<-tax[,-c(9:10)]
#head(tax)
tax<-t(tax)
tax<-as.matrix(tax) #converting tax dataframe to a matrix
#head(tax)
tax<-(t(tax)) 
tax<-phyloseq::tax_table(tax) #converting the tax matrix to phyloseq object
#class(tax)
#head(tax)
```

```{r}
# loading the ASv_counts tables
otu<-read.table("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/ASV_table.tsv", header = T, row.names=1,sep="\t")
#otu<-otu[,-c(21)]
head(otu)
names(otu)
otu<-otu_table(otu,taxa_are_rows = TRUE) #converting otu matrix to phyloseq object.
head(otu)
```

```{r}
#combine the two physeq object i.e tax,otu
physeq <-phyloseq(otu, tax)
physeq
#plot_bar(physeq, fill = "Class")
```

```{r}
#importing the metadata
meta<- read.table("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/soil_metadata.csv",sep = ",", header = TRUE, row.names = 1)
head(meta)
#names(meta) = c("Sample_ID","Animal_Type",  "Microbiome.Domain","Location")
names(meta)
head(meta)
meta<-sample_data(meta) #converting the metadata to a phyloseq object
```

```{r}
# we now create a third object called random for mergingwith the other three object
library("ape")
random_tree <- rtree(ntaxa(physeq), rooted = TRUE, tip.label = taxa_names(physeq))
#plot(random_tree)
```



```{r}
#merging the preceeding 3 objects.
physeq1 <- merge_phyloseq(physeq, meta, random_tree)
physeq1
saveRDS(physeq1, file="C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/soil_fungi.rds")
#readRDS(file ="/path/to/physeq_rumenbacteria.rds" ) #reading a .rds image
```


```{r}
#Read in Data
ps1<-readRDS("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/soil_fungi.rds")

```


```{r}
#filtering the unwanted sequences
ps2 <- subset_taxa(ps1, (Order!="Chloroplast") | is.na(Order))
ntaxa(ps2)
ps2 <- subset_taxa(ps2, (Phylum!="Chloroflexi") | is.na(Phylum))
ntaxa(ps2)
ps2<- subset_taxa(ps2, (Family!="Mitochondria") | is.na(Family))
ntaxa(ps2)
ps2<- subset_taxa(ps2, (Kingdom!="Archaea") | is.na(Kingdom))
ntaxa(ps2)
ps2<- subset_taxa(ps2, (Kingdom!="Eukaryota") | is.na(Kingdom))
ntaxa(ps2)

ps2<-subset_taxa(ps2, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))
ntaxa(ps2)
```
```{r}
ps3 <- prune_taxa(taxa_sums(ps2) > 20, ps2)
ps3
```
```{r}
#Extracting the filtered taxonomy and feature tables for barplot plotting
tax_table <- phyloseq_to_df(ps3, addtax = T, addtot = F, addmaxrank = F)
cumulation <- tax_table %>% adorn_totals(c("col"))
cumulation <- cumulation[order(cumulation$Total, decreasing = TRUE),]
```


```{r}
#merging the blast taxonomic classification to blast abundance table
merged_data <- tax_table
write.csv(merged_data, file="C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/Supp_Taxonomic_Classification.csv")
```


```{r}
#grouping the data (entire dataset): Genus, Species and sample names
Featured_table <- merged_data[,c(4,10:42)]
group <- Featured_table %>%
  group_by(Phylum)%>%
  summarise_if(is.numeric, sum)
#Groups the data in the defined order which eases downstream analysis
group <- Featured_table %>% group_by(Phylum)%>%
  summarise_each(funs(sum), "R1","R10","R12","R13","R14","R15","R16","R17","R18","R19","R2","R20","R3","R4","R5","R6","R7","R8","R9","S1","S10","S13","S14","S15","S16","S2","S3","S4","S5","S6","S7","S8","S9")
#group<-group[-c(1),]
head(group)
```
## TREATMENTS
```{r}
#creating multiple dataframes for the different treatments

ps<-group[,c(1,20,23,25,28,30,32,34)] # Push pull soil
ms<-group[,c(1,21,22,24,26,27,29,31,33)] # Monoculture soil
pr<-group[,c(1,2,5,7,9,11,13,15,17,19)] # Pushpull root
mr<-group[,c(1,3,4,6,8,10,12,14,16,18)] # Monoculture root

soil<-group[,c(1,20:34)] 
root<-group[,c(1:19)]
all<-group

```

```{r}
dim(ps)
ps_total <- ps %>% adorn_totals(c("col"))
ps_total <- mutate(ps_total, ps=rowSums(ps_total[9])/8)
ps_total <- ps_total[,c(1,10)]

dim(ms)
ms_total <- ms %>% adorn_totals(c("col"))
ms_total <- mutate(ms_total, ms=rowSums(ms_total[10])/9)
ms_total <- ms_total[,c(1,11)]

dim(pr)
pr_total <- pr %>% adorn_totals(c("col"))
pr_total <- mutate(pr_total, pr=rowSums(pr_total[11])/10)
pr_total <- pr_total[,c(1,12)]

dim(mr)
mr_total <- mr %>% adorn_totals(c("col"))
mr_total <- mutate(mr_total, mr=rowSums(mr_total[11])/10)
mr_total <- mr_total[,c(1,12)]

dim(soil)
soil_total <- soil%>% adorn_totals(c("col"))
soil_total <- mutate(soil_total, soil=rowSums(soil_total[17])/16)
soil_total <- soil_total[,c(1,18)]

dim(root)
root_total <- root %>% adorn_totals(c("col"))
root_total <- mutate(root_total, root=rowSums(root_total[20])/19)
root_total <- root_total[,c(1,21)]

```

ps_total 
ms_total 
pr_total 
mr_total 

```{r}
# Merging
##merged <- Reduce(function(x,y) merge(x,y,by="Phylum",all=TRUE),
 ##                list(ps_total,ms_total,pr_total, mr_total ))

##names(merged)<-c('Phylum','Push Soil','Mono Soil','Push Root', 'Mono Root')

merged <- Reduce(function(x,y) merge(x,y,by="Phylum",all=TRUE),
                 list(root_total,soil_total))

names(merged)<-c('Phylum','Root','Soil')

#calculating the total abundance per genus and ordering from the most abundant to the lowest
cumulation <- merged %>% adorn_totals(c("col"))
cumulation <- cumulation[order(cumulation$Total, decreasing = TRUE),]
cumulation$perc = cumulation$Total / sum(cumulation$Total) * 100

tired<-head(cumulation$Phylum, n=10)
# Using R base append()
#install.packages('rlist')
library('rlist')
li2 <- append(tired,"Others")
print(li2)
genus_Rep <- li2

group <- aggregate(merged[-1], list(Phylum = replace(merged$Phylum,!(merged$Phylum %in% genus_Rep), "Others")), sum)
#View(group) 


#PS<-group[,c(1:2)]
#MS<- group[,c(1:3)] 
#PR<- group[,c(1:4)]
#MR<- group[,c(1:5)]
All<- group[,c(1:3)]

```


# Viewing Sample Diversity
```{r}
#install.packages("janitor")
library(janitor)
#converting the abudances into percentage
bar_all <- adorn_percentages(All, denominator = "col", na.rm = T)
bar_all %>%
  adorn_totals("row") %>%
  adorn_pct_formatting()
dist_all<-bar_all %>%
  adorn_totals("row") %>%
  adorn_pct_formatting()
dist_all
write.csv(dist_all, "C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/Fungi_10_Phylum_Abundance_Sample_Type.csv")


#gathering the data
bar_all <- bar_all %>%
  gather(value = "abundance", key = "Sample_type", -Phylum)
bar_all <- as.data.frame(gsub("\\(", " (", as.matrix(bar_all)))

# coerce the dataframe columns into respective data type
bar_all$Phylum <- as.factor(bar_all$Phylum)
bar_all$Farming_type <- as.character(bar_all$Sample_type)
bar_all$abundance <- as.numeric(bar_all$abundance)

#ordering the data for plotting
bar_all$Phylum <- reorder(bar_all$Phylum, bar_all$abundance)
bar_all$Phylum <- factor(bar_all$Phylum, levels=rev(levels(bar_all$Phylum)))
bar_all$Phylum <- factor(bar_all$Phylum, 
                        levels=genus_Rep)

# Defining the color pallete
myPalette <- c("#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#99000D", "#E6AB02", "#A6761D", "#666666","#FDCDAC", "#1F78B4", "#B2DF8A", "#33A02C", "#CBD5E8", "#E31A1C", "#FDBF6F", "#FF7F00","#4A1486","#C0C0C0","#B3E2CD","#FFFF33", "#5172b2","#F4CAE4", "#E6F5C9", "#FCFBFD","#139BF1","#09FF00","#065535", "#1D91C0", "#C0FFEE","#B35806","#0C2C84","#D0ED0E","#092617","#499976","#4D5D53","#E48400","#6082B6","#316689","#CEFB02","#738678","#645452","#EEA47FFF", "#00539CFF", "#FC766AFF", "#42EADDFF", "#00A4CCFF", "#69B3BB", "#B589D6","#D1DFB7","#97BC62FF","#D198C5FF","#000000", "#CBCE91FF", "#616247FF", "#D64161FF","#435E55FF", "#DD4132FF","#CE4A7EFF", "#BD7F37FF","#FFA351FF","#185E57", "#FCE3E3", "#EF6C6C", "#EF6C9A", "#93103E", "#F7E3FC", "#D56CEF", "#BD19E6", "#8D6CEF", "#DBD1FA", "#BFC8F8", "#1531BC", "#9FA5F3", "#95C7F3", "#6CEFC8", "#6CEF84", "#91E619", "#B7CEEC", "#9AFEFF", "#57FEFF", "#78C7C7", "#46C7C7", "#00A36C", "#728C00", "#4E9268", "#6CC417", "#64E986", "#F5E216", "#FFCE44", "#8B8000", "#660000", "#610541", "#E56E94", "#F660AB", "#E3319D", "#FF77FF", "#C45AEC", "#6960EC", "#736AFF", "#F9B7FF", "#FCDFFF", "#D291BC", "#614051", "#FEA3AA", "#7D0541")

length(myPalette)

# Definig the names in italics
guide_italics <- guides(fill = guide_legend(label.theme = element_text(size = 15, 
face = "italic", colour = "Black", angle = 0)))

```

# Plotting barplot
```{r}
library(Cairo)
library(forcats)

#plotting the barplot 
p_all <- ggplot(bar_all,aes(x = fct_inorder(Sample_type), y = abundance), labs(fill= Phylum), group=row.names(bar_all))+ xlab("Sample Type")+ ylab("abundance") + geom_col(aes(fill = Phylum),position = position_stack(reverse = FALSE))+
  theme(axis.text.x = element_text(angle = 72,size = 15, hjust = 1, face = "italic", family = "Arial"))+
   scale_fill_manual(values = myPalette)+
  #guides(fill = guide_legend(reverse = FALSE))+
  guide_italics+
  theme(legend.text = element_text(size = 8, colour = "black", face = "italic", family = "Arial"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 8, family = "Arial"))+
  theme(axis.text = element_text(colour = "black", size = 8, family = "Arial"))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.1, 0.1, 0.1, 0.1, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.15, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.position = "right", legend.justification = "top", legend.direction = "vertical", legend.text = element_text(size = 10))+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 8, family = "Arial"))+
  theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 8, family = "Arial"))

#install.packages('extrafont')
library(extrafont)
#font_import
#BiocManager::install("ggpubr")

#plot4_60<-p_all + scale_y_continuous(labels = scales::percent_format(accuracy = 1)) + xlab("Site")
plot4_10<-p_all 
plot4_10
```
```{r}
ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/Fungi_10_Phylum_Abundance_Sample_Type.jpeg", 
       width = 12, height = 12, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/Fungi_10_Phylum_Abundance_Sample_Type.png", 
       width = 12, height = 12, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/Fungi_10_Phylum_Abundance_Sample_Type.svg", 
       width = 12, height = 12, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/Fungi_10_Phylum_Abundance_Sample_Type.tiff", 
       width = 12, height = 12, dpi = 600)
```


### ALPHA DIVERSITY

```{r}

ps3
set.seed(1024)
rareres <- get_rarecurve(obj=ps3, chunks=400)

p_rare <- ggrarecurve(obj=rareres,
                      indexNames=c("Observe","Chao1","ACE","Shannon"),
                      ) +
          theme(legend.spacing.y=unit(0.01,"cm"),
                legend.text=element_text(size=4))

prare1 <- ggrarecurve(obj=rareres, factorNames="Sample_type",
                      indexNames=c("Observe","Chao1","ACE","Shannon")
                      ) +
          scale_fill_manual(values=c("#89C5DA", "#DA5724", "#74D944", "#CE50CA"))+
          scale_color_manual(values=c("#89C5DA", "#DA5724", "#74D944", "#CE50CA"))+
          theme_bw()+
          theme(axis.text=element_text(size=8), panel.grid=element_blank(),
                strip.background = element_rect(colour=NA,fill="grey"),
                strip.text.x = element_text(face="bold"))          

prare2 <- ggrarecurve(obj=rareres,
                      factorNames="Sample_type",
                      shadow=FALSE,
                      indexNames=c("Observe", "Chao1", "ACE","Shannon")
                      ) +
          scale_color_manual(values=c("#89C5DA", "#DA5724", "#74D944", "#CE50CA"))+
          theme_bw()+
          theme(axis.text=element_text(size=8), panel.grid=element_blank(),
                strip.background = element_rect(colour=NA,fill="grey"),
                strip.text.x = element_text(face="bold"))
p_rare / prare1 / prare2

```


```{r}
alphaobj <- get_alphaindex(ps3)
head(as.data.frame(alphaobj))

p_alpha_ty <- ggbox(alphaobj, indexNames=c("Chao1","Pielou","Shannon","Simpson"), factorNames="Sample_type") +
           scale_fill_manual(values=c("#89C5DA", "#DA5724", "#74D944", "#CE50CA"))+
           theme(strip.background = element_rect(colour=NA, fill="grey"))
p_alpha_ty
```

```{r}
ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/Alpha_Diversity_Sample_Type.jpeg", 
       width = 18, height = 12, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/Alpha_Diversity_Sample_Type.png", 
       width = 18, height = 12, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/Alpha_Diversity_Sample_Type.svg", 
       width = 18, height = 12, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/Alpha_Diversity_Sample_Type.tiff", 
       width = 18, height = 12, dpi = 600)
```


```{r}
alphaobj <- get_alphaindex(ps3)
head(as.data.frame(alphaobj))

p_alpha_lc <- ggbox(alphaobj, indexNames=c("Chao1","Pielou","Shannon","Simpson"),factorNames="Location") +
           scale_fill_manual(values=c("#89C5DA", "#DA5724", "#74D944", "#CE50CA"))+
           theme(strip.background = element_rect(colour=NA, fill="grey"))
p_alpha_lc
```
```{r}
alphaobj <- get_alphaindex(ps3)
head(as.data.frame(alphaobj))

p_alpha_ft <- ggbox(alphaobj, indexNames=c("Chao1","Pielou","Shannon","Simpson"), factorNames="Farming_type") +
           scale_fill_manual(values=c("#89C5DA", "#DA5724", "#74D944", "#CE50CA"))+
           theme(strip.background = element_rect(colour=NA, fill="grey"))
p_alpha_ft
```


```{r}
alphaobj <- get_alphaindex(ps3)
head(as.data.frame(alphaobj))

p_alpha_fst <- ggbox(alphaobj, indexNames=c("Chao1","Pielou","Shannon","Simpson"), factorNames="FS_type") +
           scale_fill_manual(values=c("#89C5DA", "#DA5724", "#74D944", "#CE50CA"))+
           theme(strip.background = element_rect(colour=NA, fill="grey"))
p_alpha_fst
```



```{r}

ggarrange(p_alpha_ty, p_alpha_lc, p_alpha_ft, p_alpha_fst,
          ncol = 1, nrow = 4) + geom_point()
```

```{r}

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/Combined_Alpha_Diversity.jpeg", 
       width = 24, height = 36, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/Combined_Alpha_Diversity.png", 
       width = 24, height = 36, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/Combined_Alpha_Diversity.svg", 
       width =24, height = 36, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/Combined_Alpha_Diversity.tiff", 
       width = 24, height = 36, dpi = 600)

```

## Venn or Upset plot
```{r}
#BiocManager::install("eulerr")
library(eulerr)
library(microbiome)
#devtools::install_github('microsud/microbiomeutilities')
library(microbiomeutilities)

## SAMPLE TYPE
####################
ps3
table(meta(ps3)$Sample_type)

# convert to relative abundances
pseq.rel <- microbiome::transform(ps3, "compositional")


disease_states <- unique(as.character(meta(pseq.rel)$Sample_type))
print(disease_states)


list_core <- c() # an empty object to store information

for (n in disease_states){ # for each variable n in DiseaseState
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(pseq.rel, Sample_type == n) # Choose sample from DiseaseState by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from g 
                           detection = 0.0001, # 0.001 in atleast 90% samples 
                           prevalence = 0.75
                          )
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each DiseaseState.
    list_core[[n]] <- core_m # add to a list core taxa for each group.
    #print(list_core)
}

print(list_core)

# Specify colors and plot venn
# supplying colors in the order they appear in list_core
mycols <- c(root="#f5f3bb", soil="#dfa06e") 
venn_ST<-plot(venn(list_core),
     fills = mycols, main="A",legend = TRUE)

venn_ST

## LOCATION
####################

disease_states <- unique(as.character(meta(pseq.rel)$Location))
print(disease_states)


list_core <- c() # an empty object to store information

for (n in disease_states){ # for each variable n in DiseaseState
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(pseq.rel, Location == n) # Choose sample from DiseaseState by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from g 
                           detection = 0.0001, # 0.001 in atleast 90% samples 
                           prevalence = 0.75)
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each DiseaseState.
    list_core[[n]] <- core_m # add to a list core taxa for each group.
    #print(list_core)
}

print(list_core)

# Specify colors and plot venn
# supplying colors in the order they appear in list_core
mycols <- c(Bungoma="#fab3a9", Vihiga="#C6AD94", Siaya="#7fb285") 

venn_LC<-plot(venn(list_core),
     fills = mycols, main="C",legend = TRUE)
venn_LC

## FS TYPE
####################
disease_states <- unique(as.character(meta(pseq.rel)$FS_type))
print(disease_states)


list_core <- c() # an empty object to store information

for (n in disease_states){ # for each variable n in DiseaseState
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(pseq.rel, FS_type == n) # Choose sample from DiseaseState by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from g 
                           detection = 0.0001, # 0.001 in atleast 90% samples 
                           prevalence = 0.75)
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each DiseaseState.
    list_core[[n]] <- core_m # add to a list core taxa for each group.
    #print(list_core)
}

print(list_core)

# Specify colors and plot venn
# supplying colors in the order they appear in list_core

mycols <- c(PS="#7b5e7b", MS="#938274", PR="#e9eb87", MR="#B9F18C") 
venn_FST<-plot(venn(list_core),
     fills = mycols, main="D",legend = TRUE)
venn_FST



## FARMING TYPE
####################
disease_states <- unique(as.character(meta(pseq.rel)$Farming_type))
print(disease_states)


list_core <- c() # an empty object to store information

for (n in disease_states){ # for each variable n in DiseaseState
    #print(paste0("Identifying Core Taxa for ", n))
    
    ps.sub <- subset_samples(pseq.rel, Farming_type == n) # Choose sample from DiseaseState by n
    
    core_m <- core_members(ps.sub, # ps.sub is phyloseq selected with only samples from g 
                           detection = 0.0001, # 0.001 in atleast 90% samples 
                           prevalence = 0.55)
    print(paste0("No. of core taxa in ", n, " : ", length(core_m))) # print core taxa identified in each DiseaseState.
    list_core[[n]] <- core_m # add to a list core taxa for each group.
    #print(list_core)
}

print(list_core)

# Specify colors and plot venn
# supplying colors in the order they appear in list_core

mycols <- c("Push-Pull"= "#A7A2A9", "Maize-Monoculture"="#F4F7F5") 
venn_FT<-plot(venn(list_core),
     fills = mycols, main="B",legend = TRUE)
venn_FT


```


```{r}

ggarrange(venn_ST,  venn_FT, venn_LC, venn_FST,
          ncol = 2, nrow = 2) + geom_point()


ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/VENN_Diagrams.jpeg", 
       width = 24, height = 24, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/VENN_Diagrams.png", 
       width = 24, height = 24, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/VENN_Diagrams.svg", 
       width = 24, height = 24, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/VENN_Diagrams.tiff", 
       width = 24, height = 24, dpi = 600)

```

##ALPHA PART TWO
###Refer here: 
 http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/76-add-p-values-and-significance-levels-to-ggplots/

```{r}
library(dplyr)
library(tibble)
#alpha diversity estimation
physeq4 <- ps3
physeq4

#checking out the total read counts in the samples
reads <- sample_sums(physeq4)
reads

summary(sample_sums(physeq4))

library(microbiome)
#Extracting the otu table from the phyloseq object and plotting the rarefaction curve
otu_tab <- t(abundances(physeq4))
p <- vegan::rarecurve(otu_tab, 
                      step = 50, label = FALSE, 
                      sample = min(rowSums(otu_tab), 
                                   col = "blue", cex = 0.6))

set.seed(9242)  

#calculatin an even sampling depth for all the samples
rarefied <- rarefy_even_depth(physeq4, sample.size = 38)
rarefied

#calculating the alpha diversity
diversity<-alpha(rarefied, index = "all")
diversity <- rownames_to_column(diversity, "sample_id")

#Extracting the sample metadata from the phyloseq object
sdata1 <- meta(physeq4)
sdata1 <- rownames_to_column(sdata1, "sample_id")

```
## Sample Type
## CHAO1
```{r}

#Chao1 diversity estimates
chao1 <- diversity %>% select(sample_id, chao1)
chao_edited <- merge(chao1, sdata1, by = "sample_id", all = TRUE)
#chao_edited <- chao_edited[c(1:18, 62:86, 19:61),]

#confirming if the chao1 indices are normally distributed
shapiro.test(chao_edited$chao1)

#plotting chao1 distribution boxplot
Pchao <- ggboxplot(chao_edited, "Sample_type","chao1",
               color = "Sample_type", palette =c("#FBAE17", "#F52100", "#0071BC"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) + 
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0)))#+stat_compare_means()
Pchao<-Pchao + aes(x = fct_inorder(Sample_type)) + theme(legend.position = "none") + ylab("Chao1") + xlab("")
Pchao
```
## OBSERVED
```{r}

#Chao1 diversity estimates
observed <- diversity %>% select(sample_id, observed)
observed_edited <- merge(observed, sdata1, by = "sample_id", all = TRUE)
#chao_edited <- chao_edited[c(1:18, 62:86, 19:61),]

#confirming if the chao1 indices are normally distributed
shapiro.test(observed_edited$observed)

#plotting chao1 distribution boxplot
Pobs <- ggboxplot(observed_edited, "Sample_type","observed",
               color = "Sample_type", palette =c("#FBAE17", "#F52100", "#0071BC"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0)))#+ stat_compare_means()


Pobs<-Pobs + aes(x = fct_inorder(Sample_type)) + theme(legend.position = "none") + xlab("") + ylab("Observed")

Pobs
```

## SHANNON DIVERSITY
```{r}
#Extracting the shannon diversity index
shannon <- diversity %>% select(sample_id, diversity_shannon)

shannon_edited <- merge(shannon, sdata1, by = "sample_id", all = TRUE)
#shannon_edited <- shannon_edited[c(1:18, 62:86, 19:61),]

#confirming if the shannon indices are normally distributed
shapiro.test(shannon_edited$diversity_shannon)

library(ggpubr)


#plotting the boxplots for the shannon index data
P <- ggboxplot(shannon_edited, "Sample_type","diversity_shannon",
               color = "Sample_type", palette =c("#FBAE17", "#F52100", "#0071BC"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5))+
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0))) #+stat_compare_means()

Pshan<-P + aes(x = fct_inorder(Sample_type)) + theme(legend.position = "none") + xlab("") + ylab("Shannon")

Pshan
```
## Evenness

```{r}
#Extracting the shannon diversity index
even <- diversity %>% select(sample_id, evenness_pielou)

even_edited <- merge(even, sdata1, by = "sample_id", all = TRUE)
#shannon_edited <- shannon_edited[c(1:18, 62:86, 19:61),]

#confirming if the shannon indices are normally distributed
shapiro.test(even_edited$evenness_pielou)

library(ggpubr)


#plotting the boxplots for the shannon index data
P <- ggboxplot(even_edited, "Sample_type","evenness_pielou",
               color = "Sample_type", palette =c("#FBAE17", "#F52100", "#0071BC"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5))+
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0))) #+stat_compare_means()

Peven<-P + aes(x = fct_inorder(Sample_type)) + theme(legend.position = "none") + xlab(" ") + ylab("Evenness")

Peven + stat_compare_means()

```

## LOCATION
## CHAO1
```{r}

#Chao1 diversity estimates
chao1 <- diversity %>% select(sample_id, chao1)
chao_edited <- merge(chao1, sdata1, by = "sample_id", all = TRUE)
#chao_edited <- chao_edited[c(1:18, 62:86, 19:61),]

#confirming if the chao1 indices are normally distributed
shapiro.test(chao_edited$chao1)

#plotting chao1 distribution boxplot
Pchao_l <- ggboxplot(chao_edited, "Location","chao1",
               color = "Location", palette =c("#FBAE17", "#F52100", "#0071BC"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) + 
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0)))#+stat_compare_means()
Pchao_l<-Pchao_l + aes(x = fct_inorder(Location)) + theme(legend.position = "none") + xlab(" ") + ylab("Chao1")
Pchao_l
```
## OBSERVED
```{r}

#Chao1 diversity estimates
observed <- diversity %>% select(sample_id, observed)
observed_edited <- merge(observed, sdata1, by = "sample_id", all = TRUE)
#chao_edited <- chao_edited[c(1:18, 62:86, 19:61),]

#confirming if the chao1 indices are normally distributed
shapiro.test(observed_edited$observed)

#plotting chao1 distribution boxplot
Pobs_l <- ggboxplot(observed_edited, "Location","observed",
               color = "Location", palette =c("#FBAE17", "#F52100", "#0071BC"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0)))#+ stat_compare_means()


Pobs_l<-Pobs_l + aes(x = fct_inorder(Location)) + theme(legend.position = "none") + xlab(" ") + ylab("Observed")

Pobs_l
```

## SHANNON DIVERSITY
```{r}
#Extracting the shannon diversity index
shannon <- diversity %>% select(sample_id, diversity_shannon)

shannon_edited <- merge(shannon, sdata1, by = "sample_id", all = TRUE)
#shannon_edited <- shannon_edited[c(1:18, 62:86, 19:61),]

#confirming if the shannon indices are normally distributed
shapiro.test(shannon_edited$diversity_shannon)

library(ggpubr)


#plotting the boxplots for the shannon index data
P_l <- ggboxplot(shannon_edited, "Location","diversity_shannon",
               color = "Location", palette =c("#FBAE17", "#F52100", "#0071BC"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5))+
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0))) #+stat_compare_means()

Pshan_l<-P_l + aes(x = fct_inorder(Location)) + theme(legend.position = "none") + xlab(" ") + ylab("Shannon")

Pshan_l
```
## Evenness

```{r}
#Extracting the shannon diversity index
even <- diversity %>% select(sample_id, evenness_pielou)

even_edited <- merge(even, sdata1, by = "sample_id", all = TRUE)
#shannon_edited <- shannon_edited[c(1:18, 62:86, 19:61),]

#confirming if the shannon indices are normally distributed
shapiro.test(even_edited$evenness_pielou)

library(ggpubr)


#plotting the boxplots for the shannon index data
Pe_l <- ggboxplot(even_edited, "Location","evenness_pielou",
               color = "Location", palette =c("#FBAE17", "#F52100", "#0071BC"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5))+
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0))) #+stat_compare_means()

Peven_l<-Pe_l + aes(x = fct_inorder(Location)) + theme(legend.position = "none") + xlab(" ") + ylab("Evenness")

Peven_l + stat_compare_means(method = "kruskal.test", paired = TRUE)

```
## FARMING TYPE
## CHAO1
```{r}

#Chao1 diversity estimates
chao1 <- diversity %>% select(sample_id, chao1)
chao_edited <- merge(chao1, sdata1, by = "sample_id", all = TRUE)
#chao_edited <- chao_edited[c(1:18, 62:86, 19:61),]

#confirming if the chao1 indices are normally distributed
shapiro.test(chao_edited$chao1)

#plotting chao1 distribution boxplot
Pchao_ft <- ggboxplot(chao_edited, "Farming_type","chao1",
               color = "Farming_type", palette =c("#FBAE17", "#F52100", "#0071BC"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) + 
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0)))#+stat_compare_means()
Pchao_ft<-Pchao_ft + aes(x = fct_inorder(Farming_type)) + theme(legend.position = "none") + xlab(" ") + ylab("Chao1")
Pchao_ft
```
# CHAO1: FS_type
```{r}
#Chao1 diversity estimates
chao1 <- diversity %>% select(sample_id, chao1)
chao_edited <- merge(chao1, sdata1, by = "sample_id", all = TRUE)
#chao_edited <- chao_edited[c(1:18, 62:86, 19:61),]

#confirming if the chao1 indices are normally distributed
shapiro.test(chao_edited$chao1)

#plotting chao1 distribution boxplot
Pchao_fst <- ggboxplot(chao_edited, "FS_type","chao1",
               color = "FS_type", palette =c("#FBAE17", "#F52100", "#0071BC","#F4CAE4"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) + 
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0)))#+stat_compare_means()
Pchao_fst<-Pchao_fst + aes(x = fct_inorder(FS_type)) + theme(legend.position = "none") + xlab(" ") + ylab("Chao1")
Pchao_fst
```


## FARMING TYPE
## OBSERVED
```{r}

#Chao1 diversity estimates
observed <- diversity %>% select(sample_id, observed)
observed_edited <- merge(observed, sdata1, by = "sample_id", all = TRUE)
#chao_edited <- chao_edited[c(1:18, 62:86, 19:61),]

#confirming if the chao1 indices are normally distributed
shapiro.test(observed_edited$observed)

#plotting chao1 distribution boxplot
Pobs_ft <- ggboxplot(observed_edited, "Farming_type","observed",
               color = "Farming_type", palette =c("#FBAE17", "#F52100", "#0071BC"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0)))#+ stat_compare_means()


Pobs_ft<-Pobs_ft + aes(x = fct_inorder(Farming_type)) + theme(legend.position = "none") + xlab(" ") + ylab("Observed")

Pobs_ft
```

## SHANNON DIVERSITY
```{r}
#Extracting the shannon diversity index
shannon <- diversity %>% select(sample_id, diversity_shannon)

shannon_edited <- merge(shannon, sdata1, by = "sample_id", all = TRUE)
#shannon_edited <- shannon_edited[c(1:18, 62:86, 19:61),]

#confirming if the shannon indices are normally distributed
shapiro.test(shannon_edited$diversity_shannon)

library(ggpubr)


#plotting the boxplots for the shannon index data
P_ft <- ggboxplot(shannon_edited, "Farming_type","diversity_shannon",
               color = "Farming_type", palette =c("#FBAE17", "#F52100", "#0071BC"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5))+
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0))) #+stat_compare_means()

Pshan_ft<-P_ft + aes(x = fct_inorder(Farming_type)) + theme(legend.position = "none") + xlab(" ") + ylab("Shannon")

Pshan_ft
```

##FS_type

## SHANNON DIVERSITY
```{r}
#Extracting the shannon diversity index
shannon <- diversity %>% select(sample_id, diversity_shannon)

shannon_edited <- merge(shannon, sdata1, by = "sample_id", all = TRUE)
#shannon_edited <- shannon_edited[c(1:18, 62:86, 19:61),]

#confirming if the shannon indices are normally distributed
shapiro.test(shannon_edited$diversity_shannon)

library(ggpubr)


#plotting the boxplots for the shannon index data
P_fst <- ggboxplot(shannon_edited, "FS_type","diversity_shannon",
               color = "FS_type", palette =c("#FBAE17", "#F52100", "#0071BC", "#F4CAE4"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5))+
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0))) #+stat_compare_means()

Pshan_fst<-P_fst + aes(x = fct_inorder(FS_type)) + theme(legend.position = "none") + xlab(" ") + ylab("Shannon")

Pshan_fst +stat_compare_means()
```
## Evenness

```{r}
#Extracting the shannon diversity index
even <- diversity %>% select(sample_id, evenness_pielou)

even_edited <- merge(even, sdata1, by = "sample_id", all = TRUE)
#shannon_edited <- shannon_edited[c(1:18, 62:86, 19:61),]

#confirming if the shannon indices are normally distributed
shapiro.test(even_edited$evenness_pielou)

library(ggpubr)


#plotting the boxplots for the shannon index data
Pe_ft <- ggboxplot(even_edited, "Farming_type","evenness_pielou",
               color = "Farming_type", palette =c("#FBAE17", "#F52100", "#0071BC"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5))+
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0))) #+stat_compare_means()

Peven_ft<-Pe_ft + aes(x = fct_inorder(Farming_type)) + theme(legend.position = "none") + xlab(" ") + ylab("Evenness")

Peven_ft + stat_compare_means()

```

## Evenness

```{r}
#Extracting the shannon diversity index
even <- diversity %>% select(sample_id, evenness_pielou)

even_edited <- merge(even, sdata1, by = "sample_id", all = TRUE)
#shannon_edited <- shannon_edited[c(1:18, 62:86, 19:61),]

#confirming if the shannon indices are normally distributed
shapiro.test(even_edited$evenness_pielou)

library(ggpubr)


#plotting the boxplots for the shannon index data
Pe_fst <- ggboxplot(even_edited, "FS_type","evenness_pielou",
               color = "FS_type", palette =c("#FBAE17", "#F52100", "#0071BC", "#F4CAE4"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5))+
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0))) #+stat_compare_means()

Peven_fst<-Pe_fst + aes(x = fct_inorder(FS_type)) + theme(legend.position = "none") + xlab(" ") + ylab("Evenness")

Peven_fst + stat_compare_means()

```

#Label and Merge
```{r}
require(ggplot2)
require(dplyr)
require(svglite)

#Sample Type
#Pchao <-Pchao + labs(tag = "A")
#Peven<-Peven + labs(tag = "B")
#Pshan<-Pshan + labs(tag = "C")

Pchao <-Pchao + labs(tag = "A") #Sample Type
Peven<-Peven + labs(tag = "B")
Pshan<-Pshan + labs(tag = "C")

Pchao_ft <- Pchao_ft+ labs(tag = "D") #Farming Type
Peven_ft<-Peven_ft + labs(tag = "E")
Pshan_ft<-Pshan_ft + labs(tag = "F")

Pchao_l <- Pchao_l + labs(tag = "G") # Location
Peven_l<-Peven_l + labs(tag = "H")
Pshan_l<-Pshan_l + labs(tag = "I")

Pchao_fst<-Pchao_fst + labs(tag = "J") #Farming and Sample Type
Peven_fst<-Peven_fst + labs(tag = "K")
Pshan_fst<-Pshan_fst + labs(tag = "L")


#Sample Type
comb_st<-ggarrange(Pchao,  
          Peven,
          Pshan,
          ncol = 3, nrow = 1) + geom_point()
comb_st<-annotate_figure(comb_st, top = text_grob("Sample Type", 
               color = "black", face = "bold", size = 14))

# Farming Type

comb_ft<-ggarrange(Pchao_ft,  
          Peven_ft,
          Pshan_ft,
          ncol = 3, nrow = 1) + geom_point()
comb_ft<-annotate_figure(comb_ft, top = text_grob("Farming Type", 
               color = "black", face = "bold", size = 14))



# Location

comb_loc<-ggarrange(Pchao_ft,  
          Peven_ft,
          Pshan_ft,
          ncol = 3, nrow = 1) + geom_point()
comb_loc<-annotate_figure(comb_loc, top = text_grob("Location", 
               color = "black", face = "bold", size = 14))



# Farming and Sample Type

comb_fst<-ggarrange(Pchao_fst,  
          Peven_fst,
          Pshan_fst,
          ncol = 3, nrow = 1) + geom_point()
comb_fst<-annotate_figure(comb_fst, top = text_grob("Farming & Sampling Type", 
               color = "black", face = "bold", size = 14))


comb_all<-ggarrange(comb_st, comb_ft, comb_loc, comb_fst,
          ncol = 1, nrow = 4) + geom_point()


#comb_all<-annotate_figure(comb_all, top = text_grob("Alpha Diversity", 
          #     color = "black", face = "bold", size = 14))
comb_all

```
```{r}
ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/Annotated_Alpha_Diversity.jpeg",
       width = 36, height = 36, dpi = 600)
ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/Annotated_Type_Alpha_Diversity.png",
       width = 36, height = 36, dpi = 600)
ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/Annotated_Type_Alpha_Diversity.svg",
       width = 36, height = 36, dpi = 600)
ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/Annotated_Type_Alpha_Diversity.tiff",
       width = 36, height = 36, dpi = 600)

```

## BETA DIVERSITY


```{r}
#ordinating the phyloseq object
library(vegan)
library(ggpubr)
#weighted unifrac
Palette <- c('#89C5DA', "#DA5724", "#74D944", "#CE50CA", "#3F4921", "#C0717C", "#CBD588", "#5F7FC7", "#673770", "#D3D93E", "#38333E", "#508578", "#D7C1B1", "#689030", "#AD6F3B", "#CD9BCD", "#D14285", "#6DDE88", "#652926", "#7FDCC0", "#C84248", "#8569D5", "#5E738F", "#D1A33D", "#8A7C64", "#599861")

ordu = ordinate(ps3, "PCoA", "unifrac", weighted = TRUE)
p <- plot_ordination(ps3, ordu, color="Sample_type")+ geom_point(size=2) +
  scale_color_manual(values = Palette) + 
  theme(axis.text.x = element_text(angle = 0, hjust = 1))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.text = element_text(size = rel(1), colour = "black"))+
  theme(legend.title = element_text(face = NULL))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = rel(1)))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))

bA1<-p + stat_ellipse()  + labs(tag = "A")


p <- plot_ordination(ps3, ordu, color="Farming_type")+ geom_point(size=2) +
  scale_color_manual(values = Palette) + 
  theme(axis.text.x = element_text(angle = 0, hjust = 1))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.text = element_text(size = rel(1), colour = "black"))+
  theme(legend.title = element_text(face = NULL))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = rel(1)))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))

bB1<-p + stat_ellipse()  + labs(tag = "B")


p <- plot_ordination(ps3, ordu, color="Location")+ geom_point(size=2) +
  scale_color_manual(values = Palette) + 
  theme(axis.text.x = element_text(angle = 0, hjust = 1))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.text = element_text(size = rel(1), colour = "black"))+
  theme(legend.title = element_text(face = NULL))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = rel(1)))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))

bC1<-p + stat_ellipse()  + labs(tag = "C")


p <- plot_ordination(ps3, ordu, color="FS_type")+ geom_point(size=2) +
  scale_color_manual(values = Palette) + 
  theme(axis.text.x = element_text(angle = 0, hjust = 1))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.text = element_text(size = rel(1), colour = "black"))+
  theme(legend.title = element_text(face = NULL))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = rel(1)))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))

bD1<-p + stat_ellipse()  + labs(tag = "D")



require(ggplot2)
require(dplyr)
require(svglite)
ggarrange(bA1, bB1, bC1, bD1,
          ncol = 2, nrow = 2) + geom_point()

```

```{r}

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/Beta_Diversity_Added.jpeg", 
       width = 24, height = 24, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/Beta_Diversity_Added.png", 
       width = 24, height = 24, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/Beta_Diversity_Added.svg", 
       width = 24, height = 24, dpi = 600)

ggsave("C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/Beta_Diversity_Added.tiff", 
       width = 24, height = 24, dpi = 600)

```

## Beta Analysis

##  Permutational Multivariate Analysis of Variance

## SAMPLE TYPE
##  Permutational Multivariate Analysis of Variance
```{r}
distme <- get_dist(ps3, distmethod ="bray", method="hellinger")
sampleda <- data.frame(sample_data(ps3), check.names=FALSE)
sampleda <- sampleda[match(colnames(as.matrix(distme)),rownames(sampleda)),,drop=FALSE]
sampleda$Sample_type <- factor(sampleda$Sample_type)
set.seed(1024)
adores <- adonis2(distme ~ Sample_type, data=sampleda, permutation=9999)
perm<-data.frame(adores)
write.csv(perm, "C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/ITS_Permutational_Multivariate_Analysis_Sampling_type.csv")
```


## FARMING TYPE
##  Permutational Multivariate Analysis of Variance
```{r}
distme <- get_dist(ps3, distmethod ="bray", method="hellinger")
sampleda <- data.frame(sample_data(ps3), check.names=FALSE)
sampleda <- sampleda[match(colnames(as.matrix(distme)),rownames(sampleda)),,drop=FALSE]
sampleda$Farming_type <- factor(sampleda$Farming_type)
set.seed(1024)
adores <- adonis2(distme ~ Farming_type, data=sampleda, permutation=9999)
perm<-data.frame(adores)
write.csv(perm, "C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/ITS_Permutational_Multivariate_Analysis_Farming_type.csv")
```


## LOCATION
##  Permutational Multivariate Analysis of Variance
```{r}
distme <- get_dist(ps3, distmethod ="bray", method="hellinger")
sampleda <- data.frame(sample_data(ps3), check.names=FALSE)
sampleda <- sampleda[match(colnames(as.matrix(distme)),rownames(sampleda)),,drop=FALSE]
sampleda$Location <- factor(sampleda$Location)
set.seed(1024)
adores <- adonis2(distme ~ Location, data=sampleda, permutation=9999)
perm<-data.frame(adores)
write.csv(perm, "C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/ITS_Permutational_Multivariate_Analysis_Location.csv")
```

## FStype
##  Permutational Multivariate Analysis of Variance
```{r}
distme <- get_dist(ps3, distmethod ="bray", method="hellinger")
sampleda <- data.frame(sample_data(ps3), check.names=FALSE)
sampleda <- sampleda[match(colnames(as.matrix(distme)),rownames(sampleda)),,drop=FALSE]
sampleda$FS_type <- factor(sampleda$FS_type)
set.seed(1024)
adores <- adonis2(distme ~ FS_type, data=sampleda, permutation=9999)
perm<-data.frame(adores)
write.csv(perm, "C:/Users/icipe/Desktop/active_manuscript/soil/dada2_ITS/finale_results/pub/ITS_Permutational_Multivariate_Analysis_FS_type.csv")
```

